src/languages/regex.js

```diff
@@ -0,0 +1,101 @@
+/*
+Language: regex
+Description: Regular expressions 
+Author: Konrad Rudolph <konrad.rudolph@gmail.com>
+*/
+
+export default function (hljs) {
+  const ESCAPE_SEQUENCE = {
+    className: 'built_in',
+    begin: [
+      // Unicode script/block/category
+      /\\[pP](?:[^{]|\{\^?[^}]+\})/,
+      // Unicode character
+      /\\[ux](?:[0-9a-fA-F]{1,4}|\{[0-9a-fA-F]{1,4}\})/,
+      // Backreferences
+      /\\k(?:'-?\d+'|<-?\d+>|{-?\d+})/,
+      /\\g(?:'-?\d+'|<-?\d+>|{-?\d+}|-?\d+)/,
+      // Everything else
+      /\\./
+    ].map(r => r.source).join('|')
+  };
+
+  const CHARACTER_CLASS = {
+    className: 'string',
+    begin: /\[\^?]?/,
+    end: /]/,
+    contains: [
+      {
+        className: 'literal',
+        variants: [
+          // POSIX named character class.
+          { begin: /\[:\^?/, end: /:]/ },
+          // POSIX collation sequences.
+          { begin: /\[\./, end: /\.]/ },
+          // POSIX character equivalence.
+          { begin: /\[=/, end: /=]/ }
+        ]
+      },
+      ESCAPE_SEQUENCE
+    ]
+  };
+
+  const META = {
+    className: 'keyword',
+    begin: /[.+*?|^$]|\{(?:\d+(?:,\d*)?|,\d+)\}/,
+  };
+
+  const COMMENT = hljs.COMMENT(/\(\?#/, /\)/);
```

Comment:
Regex can have comments? LOL...  now we have a clear conflict with `regex.comment` vs `comment`. :-)